src/Common/QueryProfiler.cpp

```diff
@@ -120,6 +120,10 @@ QueryProfilerBase<ProfilerImpl>::QueryProfilerBase(const UInt64 thread_id, const
     if (sigaction(pause_signal, &sa, previous_handler))
         throwFromErrno("Failed to setup signal handler for query profiler", ErrorCodes::CANNOT_SET_SIGNAL_HANDLER);
 
+#if defined(OS_LINUX)
+    write_trace_iteration = 0;
```

Comment:
Does it change anything?

Reply:
There is some heuristics based on this counter, to not skip profiling completely on overruns, https://github.com/ClickHouse/ClickHouse/blob/e5b811f96d0adc0897f150124b487d6fb46fdc9c/src/Common/QueryProfiler.cpp#L45

Reply:
Ok but:
- the heuristic is to write every n-th event, does not matter what exactly;
- only some incremental counter is needed, it does not matter what is the initial value.

Reply:
Yep, it is merely the same, Agree, I will close this.